src/main/java/com/fasterxml/jackson/databind/util/LookupCache.java

```diff
@@ -0,0 +1,29 @@
+package com.fasterxml.jackson.databind.util;
+
+import com.fasterxml.jackson.core.util.Snapshottable;
+
+import java.util.function.BiConsumer;
+
+/**
+ * An interface describing the required API for the Jackson-Databind Type cache.
+ *
+ * @see com.fasterxml.jackson.databind.type.TypeFactory#withCache(LookupCache<java.lang.Object,com.fasterxml.jackson.databind.JavaType>)
+ * @see SimpleLookupCache
+ */
+public interface LookupCache <K,V>
+        extends Snapshottable<LookupCache<K,V>>,
+        java.io.Serializable {
+    void contents(BiConsumer<K,V> consumer);
```

Comment:
Would be good to add Javadocs in general. But especially here, as to if and why iteration over entries is desired for all implementations.

Reply:
ReadOnlyClassToSerializerMap relies on this being implemented (only usage I found) - https://github.com/FasterXML/jackson-databind/blob/master/src/main/java/com/fasterxml/jackson/databind/ser/impl/ReadOnlyClassToSerializerMap.java#L36

Reply:
Ah. Hmmh. Yes, that makes sense then... if that specific case is to be made pluggable.

Actually: let's drop that from interface, and not make this particular cache pluggable.
So `contents()` is perfectly fine for `SimpleLookupCache` for now. We can consider adding it into interface later on if other use is needed, but for now I'd rather keep API minimalistic.